frontend/js/pages/BattleCreate.js

```diff
@@ -0,0 +1,65 @@
+import { Formik, Field, Form } from 'formik';
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+
+import { createBattleAction } from '../actions/createBattle';
+import { getCurrentUser } from '../actions/getUser';
+
+function BattleCreate(props) {
+  const { user, errorMessage } = props;
+
+  useEffect(() => {
+    if (!user) {
+      props.getCurrentUser();
+    }
+  }, []);
+
+  if (user) {
+    return (
+      <div className="battle_container_detail">
+        <Formik
```

Comment:
let's add some validations of this form? You can use https://github.com/jquense/yup or use the default validations provided by formik

Reply:
great!